src/test/java/org/jenkinsci/plugins/credentialsbinding/masking/AlmquistShellSecretPatternFactoryTest.java

```diff
@@ -0,0 +1,140 @@
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+import java.util.UUID;
+
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.not;
+import static org.hamcrest.Matchers.startsWith;
+import static org.jenkinsci.plugins.credentialsbinding.test.Executables.executable;
+import static org.junit.Assume.assumeThat;
+
+@RunWith(Theories.class)
+public class AlmquistShellSecretPatternFactoryTest {
+
+    // ' is escaped as '"'"', '' as '"''"', ''' as '"'''"'
+    public static final @DataPoint String MULTIPLE_QUOTES = "ab'cd''ef'''gh";
+    // "starting" and "ending" single quotes are escaped as "middle" single quotes
+    public static final @DataPoint String SURROUNDED_BY_QUOTES = "'abc'";
+    public static final @DataPoint String SURROUNDED_BY_QUOTES_AND_MIDDLE = "'ab'cd'";
+    public static final @DataPoint String SIMPLE_CASE_1 = "abc";
+    public static final @DataPoint String SIMPLE_CASE_2 = "ab'cd";
+    public static final @DataPoint String SIMPLE_CASE_3 = "ab''cd";
+    public static final @DataPoint String SIMPLE_CASE_4 = "ab'c'd";
+    public static final @DataPoint String LEADING_QUOTE = "'a\"b\"c d";
+    public static final @DataPoint String TRAILING_QUOTE = "a\"b\"c d'";
+    public static final @DataPoint String SAMPLE_PASSWORD = "}#T14'GAz&H!{$U_";
+    public static final @DataPoint String ANOTHER_SAMPLE_PASSWORD = "a'b\"c\\d(e)#";
+    public static final @DataPoint String ONE_MORE = "'\"'(foo)'\"'";
+    public static final @DataPoint String FULL_ASCII = "!\"#$%&'()*+,-./ 0123456789:;<=>? @ABCDEFGHIJKLMNO PQRSTUVWXYZ[\\]^_ `abcdefghijklmno pqrstuvwxyz{|}~";
+
+    @DataPoints
+    public static List<String> generatePasswords() {
+        Random random = new Random(100);
+        List<String> passwords = new ArrayList<>(10);
+        for (int i = 0; i < 10; i++) {
+            int length = random.nextInt(24) + 8;
+            StringBuilder sb = new StringBuilder(length);
+            for (int j = 0; j < length; j++) {
+                // choose a (printable) character in the closed range [' ', '~']
+                // 0x7f is DEL, 0x7e is ~, and space is the first printable ASCII character
+                char next = (char) (' ' + random.nextInt('\u007f' - ' '));
+                sb.append(next);
+            }
+            passwords.add(sb.toString());
+        }
+        return passwords;
+    }
+
+    @ClassRule public static JenkinsRule j = new JenkinsRule();
+
+    private WorkflowJob project;
+    private String credentialsId;
+
+    @BeforeClass
+    public static void assumeBash() {
+        // ash = Almquist shell, default one used in Alpine
+        assumeThat("ash", is(executable()));
+        // due to https://github.com/jenkinsci/durable-task-plugin/blob/e75123eda986f20a390d92cc892c3d206e60aefb/src/main/java/org/jenkinsci/plugins/durabletask/BourneShellScript.java#L149
+        // on Windows
+        assumeThat("nohup", is(executable()));
```

Comment:
Huh? Why is this here? The link does not explain anything to me.

Reply:
The link points to a code adding the `nohup` argument in the command. On Windows that command is not necessary present (at least not in my environments) and thus just checking for the shell is not sufficient when used in `durable-task`.

Reply:
I would not expect `ash` to be present on Windows either.

Reply:
Good point 😕

Reply:
Anyway, it is OK, just a bit confusing.